src/optimizer/pushdown/pushdown_cross_product.cpp

```diff
@@ -32,10 +33,59 @@ unique_ptr<LogicalOperator> FilterPushdown::PushdownCrossProduct(unique_ptr<Logi
 			}
 		}
 	}
+
+	//! Check if columns in join condition have filters and push them down to both tables.
+	vector<unique_ptr<Filter>> filters_add_right;
```

Comment:
This seems to perform transitive filters based on join conditions, correct? The filters here should come from a FilterCombiner, and thus transitive filters should already have been added. If any filters have been missed could you look at the FilterCombiner to see why these filters have not been added correctly? That could avoid having to re-implement transitive filters here again.